retrospective-update/functions.py

```
@@ -0,0 +1,243 @@
+
+
+def concatenate_outputs(outputs_dir: str,
+                        hourly_zarr_path: str,
+                        daily_zarr_path: str,
+                        CL: CloudLog) -> None:
+    # Build the week dataset
+    # for each unique start date, sorted in order, open/merge the files from all vpus and append to the zarr
+    vpu_outputs = natsorted(glob(os.path.join(outputs_dir, '*')))
+    unique_outputs = natsorted(glob(os.path.join(vpu_outputs[0], '*')))
+    if not unique_outputs:
+        CL.ping('FAIL', f"No-Qout-files-found")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)